src/test/java/org/broadinstitute/hellbender/tools/walkers/coverage/DepthOfCoverageIntegrationTest.java

```diff
@@ -0,0 +1,259 @@
+        final File output = IOUtils.createTempFileInDirectory( "testNoCoverageDueToFiltering", ".csv", baseOutputFile);
+
+        String[] cmd = new String[]{ "-R",hg38Reference,
+                "-I",largeFileTestDir+"multiSampleSubsetted.bam",
+                "-L", "chr1:1656275-1677440", // this should only completely span two of the genes in the list
+                "--calculate-coverage-over-genes",getTestFile("refGene_CDK11B.refseq").getAbsolutePath(),
+                "--min-base-quality","0","-dels","-pt","sample","--output-format","CSV","--omit-depth-output-at-each-base","--omit-per-sample-statistics","--omit-genes-not-entirely-covered-by-traversal",
+                "-O",output.getAbsolutePath()};
+        runCommandLine(cmd);
+
+        try (final CSVReader actualParser = new CSVReader(new FileReader(output.getAbsolutePath()+".sample_gene_summary"))) {
+            actualParser.readNext(); // Skip header line
+            List<String[]> lines = actualParser.readAll();
+            Assert.assertEquals(lines.size(), 2);
+            // The only two genes completely encompassed by this interval span
+            Assert.assertEquals(lines.get(0)[0], "SLC35E2");
+            Assert.assertEquals(lines.get(1)[0], "SLC35E2");
+            Assert.assertNotEquals(lines.get(0), lines.get(1));
+        }
+    }
+
+    @Test
+    // NOTE, the gene list file was not generated with GATK3 due to different gene list merging behavior
+    public void testGeneListDataPartallylCoveredIntronGap() throws IOException {
+        final File baseOutputFile = createTempDir("testNoCoverageDueToFiltering");
+        final File output = IOUtils.createTempFileInDirectory( "testNoCoverageDueToFiltering", ".csv", baseOutputFile);
+
+        String[] cmd = new String[]{ "-R",hg38Reference,
+                "-I",largeFileTestDir+"multiSampleSubsetted.bam",
+                "-L", "chr1:1656275-1666275", "-L", "chr1:1666280-1677440", // this has a gap that aligns with introns for both genes
+                "--calculate-coverage-over-genes",getTestFile("refGene_CDK11B.refseq").getAbsolutePath(),
+                "--min-base-quality","0","-dels","-pt","sample","--output-format","CSV","--omit-depth-output-at-each-base","--omit-per-sample-statistics","--omit-genes-not-entirely-covered-by-traversal",
+                "-O",output.getAbsolutePath()};
+        runCommandLine(cmd);
+
+        try (final CSVReader actualParser = new CSVReader(new FileReader(output.getAbsolutePath()+".sample_gene_summary"))) {
+            actualParser.readNext(); // Skip header line
+            List<String[]> lines = actualParser.readAll();
+            Assert.assertEquals(lines.size(), 2);
+            // The only two genes completely encompassed by this interval span
+            Assert.assertEquals(lines.get(0)[0], "SLC35E2");
+            Assert.assertEquals(lines.get(1)[0], "SLC35E2");
+            Assert.assertNotEquals(lines.get(0), lines.get(1));
+        }
```

Comment:
Here too you're not calling compareOutputDirectories() to fully check the output files?